### ISSUE TITLE
com.datastax.oss.driver.api.core.NoNodeAvailableException: No node was available to execute the query

### ISSUE THREAD

**Issue by dinchand** [ing-bank/cassandra-jdbc-wrapper#35]
I am trying to use the new version of liquibase-cassandra which uses this driver. However I keep getting the `com.datastax.oss.driver.api.core.NoNodeAvailableException: No node was available to execute the query` So created a small snippet only with the driver which also results in the same error.
```
public class HelloCassandra {
    public static void main(final String[] args) throws SQLException {
        // Used driver: com.ing.data.cassandra.cassandra.jdbc.CassandraDriver
        final String url = "jdbc:cassandra://node1.dc1-llds-1-tst.cassandra.xx:9142/contacting?configfile=/Users/BO55NK/IdeaProjects/untitled/src/main/resources/test.conf";
       
        final Connection connection = DriverManager.getConnection(url);
        connection.prepareStatement("Select * from contacting.contacting_configs").executeQuery();
    }
}
```

My test.conf looks like this

```
datastax-java-driver {

  basic.load-balancing-policy {
    class = DefaultLoadBalancingPolicy
    local-datacenter: DC1
  }
  basic.session-keyspace = "contacting"
  basic.request {
    timeout = 20 seconds
    consistency = LOCAL_ONE

  }
  advanced.auth-provider {
    class = PlainTextAuthProvider
    username: xxx_owner
    password: "xxxxxxx"
  }
  advanced.ssl-engine-factory {
    class = DefaultSslEngineFactory
    hostname-validation = false

    truststore-path = /resources/http-trust.jks
    truststore-password = xxxx
  }
}
```
What could I be possibly missing ?

**Comment by maximevw** [comment]
Hello @dinchand,

I found a serious lead explaining the issue and it's neither related to SSL nor your configuration (which seems totally valid).

When a configuration file is present, the JDBC wrapper discards a lot of other parameters, including `localdatacenter` because for this one it should rely on the parameter `basic.load-balancing-policy.local-datacenter` from the configuration file. But when I checked the driver context in the session in this specific case, the property `localDataCentersFromBuilder` was not set as expected.

But I found this in the documentation of the configuration file about `local-datacenter` property:
> This can also be specified programmatically with SessionBuilder.withLocalDatacenter. If both are specified, the programmatic value takes precedence.

Also the current code of the JDBC wrapper always call `SessionBuilder.withLocalDatacenter()` even if a configuration file is used and the `localdatacenter` parameter is discarded. So, the local data center used in the load balancing policy becomes empty and the queries failed with the `NoNodeAvailableException`.

I'm working on a patch to solve this, stay tuned.
